packages/@aws-cdk/aws-cloudfront/lib/response-headers-policy.ts

```diff
@@ -140,6 +140,14 @@ export class ResponseHeadersPolicy extends Resource implements IResponseHeadersP
       xssProtection: behavior.xssProtection,
     };
   }
+
+  private generateUniqueId(): string {
```

Comment:
This pattern exists in a few places. It might be nice to have `Names.uniqueId` take an optional second parameter for the maximum allowed length. That way, we could get rid of all these `generateUniqueId` methods and just say `Names.uniqueId(this, 128)`.

Reply:
it would be better to use `Names.uniqueResourceName()` instead. It has length handling built in.

Reply:
Hmm, I can't seem to find that `uniqueResourceName` anywhere with a quick grep.

Reply:
Here you go: https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/core/lib/names.ts#L84

Reply:
Doh, my fork was out of date 😅 - thank you!

Reply:
updated with [`63bc9b6` (#21525)](https://github.com/aws/aws-cdk/pull/21525/commits/63bc9b63741f9029cc39d3b81488d28451a66b89)